src/Dialog/Dialog.js

```diff
@@ -172,9 +173,12 @@ class Dialog extends React.Component<ProvidedProps & Props> {
     transition: Fade,
   };
 
+  // TODO: Clean up
+  labelledById = 'MuiDialogTitle-' + Math.random().toString(36).substr(2, 10);
```

Comment:
Here we are, we can't use a random number with server side rendering. There is no way the server add client output will match for two consecutives rendering. We are lucky that the portal doesn't support SSR yet. But that might come. This why I said it's hard. Also, there is a little overhead.

Reply:
I agree we shouldn't build like this if there's a good chance we will support Portal SSR support in future. 

Not sure if there's another way...